Wikipedia/assets/codemirror/codemirror-index.html

```diff
@@ -358,8 +358,10 @@
         buttonsToDisable.push(buttonPayload('clearFormatting'))
       }
 
-      const tokenType = doc.getTokenTypeAt(doc.getCursor())
-      if ((tokenType && tokenType.length > 0) || selectedButtons.length > 0) {
+      const cursorPosition = doc.getCursor()
+      const token = doc.getTokenAt(cursorPosition)
+      const isAtLineEnd = doc.getLine(cursorPosition.line).length === cursorPosition.ch
+      if (((token.type && token.type.length > 0) || selectedButtons.length > 0) && !isAtLineEnd) {
```

Comment:
Let me know if there's a better way to check if a cursor is placed inside wikitext. In `selectedButtons`, we store only the buttons that we care about in the UI so checking that array is not enough. When the cursor is at the end of the line, `getTokenAt` will return the adjacent token so checking token alone is not enough - that's why there's an additional line check.

Reply:
It looks like the range determination bundle could be used for this. I modified it a bit to detect ranges of template argument names (prior to that it would allow media insertion in the middle of a template argument name) and it seems to behave mostly as expected. One case that doesn't work is a line with an unterminated template on it like `{{Infobox coin` in today's featured article. Feel free to revert or update if there's something I'm missing.